### PR TITLE
ci: add release pipeline

### PR DESCRIPTION
**Pull request by DeepakBomjan**

### CODE REVIEW

**Review by debendraoli — CHANGES_REQUESTED** [review]
Please add centralized relay specific information.

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - darwin
+    goarch:
+      - amd64
+    env:
+      - CC=o64-clang
+      - CXX=o64-clang++
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X github.com/cosmos/relayer/v2/cmd.Version={{ .Tag }}
```
Comment (debendraoli) [inline .goreleaser.yaml:16]: Here. centralized relay

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      - darwin
+    goarch:
+      - arm64
+    env:
+      - CC=oa64-clang
+      - CXX=oa64-clang++
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X github.com/cosmos/relayer/v2/cmd.Version={{ .Tag }}
```
Comment (debendraoli) [inline .goreleaser.yaml:30]: Same thing.

File: .goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      - linux
+    goarch:
+      - amd64
+    env:
+      - CC=x86_64-linux-gnu-gcc
+      - CXX=x86_64-linux-gnu-g++
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X github.com/cosmos/relayer/v2/cmd.Version={{ .Tag }}
```
Comment (debendraoli) [inline .goreleaser.yaml:44]: Same goes here

File: .goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      - linux
+    goarch:
+      - arm64
+    env:
+      - CC=aarch64-linux-gnu-gcc
+      - CXX=aarch64-linux-gnu-g++
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X github.com/cosmos/relayer/v2/cmd.Version={{ .Tag }}
```
Comment (debendraoli) [inline .goreleaser.yaml:58]: Same goes here

Comment (debendraoli) [reply .goreleaser.yaml:58]: Please version information too.
cmd.VERSION

**Review by debendraoli — CHANGES_REQUESTED** [review]

File: .goreleaser.yaml (hunk 1 above)
Comment (DeepakBomjan) [reply .goreleaser.yaml]: Corrected.

File: .goreleaser.yaml (hunk 3 above)
Comment (DeepakBomjan) [reply .goreleaser.yaml]: Corrected.

File: .goreleaser.yaml (hunk 4 above)
Comment (DeepakBomjan) [reply .goreleaser.yaml]: Corrected.

**Review by debendraoli — APPROVED** [review]

**Review by debendraoli — APPROVED** [review]

**Pull request merged**